Content.Server/Atmos/Piping/Unary/EntitySystems/GasThermoMachineSystem.cs

```diff
@@ -49,82 +50,90 @@ private void OnThermoMachineUpdated(EntityUid uid, GasThermoMachineComponent the
 
             if (!MathHelper.CloseTo(combinedHeatCapacity, 0, 0.001f))
             {
-                _appearance.SetData(uid, ThermoMachineVisuals.Enabled, true);
                 var combinedEnergy = thermoMachine.HeatCapacity * thermoMachine.TargetTemperature + airHeatCapacity * inlet.Air.Temperature;
                 inlet.Air.Temperature = combinedEnergy / combinedHeatCapacity;
             }
-
-            // TODO ATMOS: Active power usage.
         }
 
-        private void OnThermoMachineLeaveAtmosphere(EntityUid uid, GasThermoMachineComponent component, AtmosDeviceDisabledEvent args)
+        private void OnGasThermoRefreshParts(EntityUid uid, GasThermoMachineComponent thermoMachine, RefreshPartsEvent args)
         {
-            _appearance.SetData(uid, ThermoMachineVisuals.Enabled, false);
```

Comment:
Looks like this got removed?

I'm not familiar enough with atmos code to comment on some of these changes and not sure if stuff breaks.

Reply:
The original code was :

```c#
        private void OnThermoMachineLeaveAtmosphere(EntityUid uid, GasThermoMachineComponent component, AtmosDeviceDisabledEvent args)
        {
            _appearance.SetData(uid, ThermoMachineVisuals.Enabled, false);

            DirtyUI(uid, component);
        }
```

So basically just appearance/UI updating (that is now  handled by the `ApcPowerReceiver` component).

> I'm not familiar enough with atmos code to comment on some of these changes and not sure if stuff breaks.

I've not touched any of the actual atmos logic (located in `OnThermoMachineUpdated`) and tested that at the best of my abilities (i.e in local), but i fully understand your stance (i don't know if or who i should ping though).

Thanks for the re-review anyway.